ludwig/trainers/trainer.py

```diff
@@ -467,70 +473,66 @@ def tune_batch_size(
     ) -> int:
         logger.info("Tuning batch size...")
 
-        if self.is_cpu_training():
-            # TODO(geoffrey): add support for batch size tuning on CPU
-            logger.warn("Batch size tuning is not supported on CPU")
-            # batch size will default to max_batch_size if it is set
-            if self.max_batch_size < DEFAULT_BATCH_SIZE:
-                logger.warn(f"Falling back to max_batch_size config param: {self.max_batch_size}")
-                best_batch_size = self.max_batch_size
-            else:
-                logger.warn(f"Falling back to default batch size: {DEFAULT_BATCH_SIZE}")
-                best_batch_size = DEFAULT_BATCH_SIZE
-        else:
+        def _is_valid_batch_size(batch_size):
+            # make sure that batch size is valid (e.g. less than size of ds)
+            is_smaller_than_training_set = batch_size < len(training_set)
+            is_under_max_batch_size = batch_size <= self.max_batch_size
+            is_valid = is_smaller_than_training_set and is_under_max_batch_size
+            if not is_valid:
+                logger.info(
+                    f"Batch size {batch_size} is invalid, must be smaller than training set size "
+                    f"{len(training_set)} and less than or equal to max batch size {self.max_batch_size}"
+                )
+            return is_valid
+
+        # TODO (ASN) : Circle back on how we want to set default placeholder value
+        # Currently, since self.batch_size is originally set to auto, we provide a
+        # placeholder starting value
+        batch_size = 2
```

Comment:
nit: is there anyway to support `batch_size=1`?

Reply:
No, because of batch norm. We skip all batches with batch_size<2 during training.

Reply:
Ah good to know, glad I asked. I'll have to update my batch size halving PR to avoid this case then. Thanks!